### ISSUE TITLE
--warn-unused-ignores in non-checked branches

### ISSUE THREAD

**Issue by srittau** [python/mypy#5940]
Consider the following mixed Python 2 and 3 stub:

```python
import sys

if sys.version_info < (3,):
    class X:
        def foo(self) -> int: ...
    class Y(X):
        def foo(self) -> str: ...  # type: ignore
```

Using mypy 0.641 and Python 3.6.7 when running this with `mypy --warn-unused-ignores` mypy wll complain:

```
foo.pyi:7: note: unused 'type: ignore' comment
```

As expected it will not complain when running in `--py2` mode. I believe mypy should not complain about ignores in branches that are not checked.

See also #2961, which is more generic.

**Comment by posita** [comment]
This isn't limited to 2 vs. 3. There are interface complications among versions of 3. This is a contrived example, but it gets the point across:

``` python
from abc import abstractmethod
try:
    from typing import Protocol, runtime_checkable
except ImportError:
    from typing_extensions import Protocol, runtime_checkable

@runtime_checkable
class _SupportsFloorCeil(Protocol):
    @abstractmethod
    def __floor__(self) -> int:
        pass
    @abstractmethod
    def __ceil__(self) -> int:
        pass

import sys
val: _SupportsFloorCeil
_default: float = 1.6180339887
if sys.version_info < (3, 9):
    val = _default  # type: ignore [assignment]  # float doesn't have __floor__ or __ceil__ prior to 3.9
else:
    val = _default
```

**Comment by NeilGirdhar** [comment]
I just ran into this [here](https://github.com/NeilGirdhar/tjax/blob/master/tjax/_src/dataclasses/helpers.py#L40
).

**Comment by ilevkivskyi** [comment]
This is a duplicate of https://github.com/python/mypy/issues/8823 (or rather vice versa, but that issue has more discussion).
